sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -84,24 +86,125 @@ static RT::PiProgram createSpirvProgram(const RT::PiContext Context,
   return Program;
 }
 
-RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
-                                                    const context &Context) {
+DeviceImage &ProgramManager::getDeviceImage(OSModuleHandle M,
+                                            const string_class &KernelName,
+                                            const context &Context) {
+  if (DbgProgMgr > 0)
+    std::cerr << ">>> ProgramManager::getDeviceImage(" << M << ", \""
+              << KernelName << "\", " << getRawSyclObjImpl(Context) << ")\n";
+
+  KernelSetId KSId = getKernelSetId(M, KernelName);
+  return getDeviceImage(M, KSId, Context);
+}
+
+static bool is_device_binary_type_supported(const context &C,
```

Comment:
Why the check is based on a context instead of device?

Reply:
Looks like this should check the device instead. Since this code was only moved by this patch I would rather address this in a separate commit along with the rest of the comments regarding `isDeviceBinaryTypeSupported`.

Reply:
Ok.